File: flick.js
//The function's parameters is an array of strings that may or may not include a specific 'toggle' word which is 'flick'
//The function must return True for each string in the array as a default. But if the string is 'flick', then the default boolean for 'flick' and every subsequent word after it is False until there is another instance of 'flick'.
//['dog', 'cat', 'bird', 'flick', 'snow', 'flick'] - [true, true, true, false, false, true]
//Create an empty array and set a default variable to true. Then run a for loop on the array. If the iterating variable is equal to 'flick', set the default boolean to the opposite of the current default. Push that boolean into the new array. Then check if the iterating variable is just a regular string, then just push that boolean into the array.


function flickSwitch(arr){
  let newArr = []
  let x = true
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === 'flick'){
      x = !x
      newArr.push(x)
    } else if (arr[i]){
      newArr.push(x)
    }
  }
  return newArr
}

console.log(flickSwitch(["flick", "flick", "flick", "flick", "flick"]))
console.log(flickSwitch(["bird"]))
console.log(flickSwitch(["flick", "mountain", "tiger", "flick", "dog"]))
console.log(flickSwitch(["plane", "jar", "bed", "flick", "boat"]))
console.log(flickSwitch(["flick"]))

File: flick.py
#The function's parameters is a list of strings that may or may not include a specific 'toggle' word which is 'flick'
#The function must return True for each string in the list as a default. But if the string is 'flick', then the default boolean for 'flick' and every subsequent word after it is False until there is another instance of 'flick'.
#['dog', 'cat', 'bird', 'flick', 'snow', 'flick'] - [True, True, True, False, False, True]
#Create an empty list and set a default variable to True. Then run a for loop on the array. If the iterating variable is equal to 'flick', set the default boolean to the opposite of the current default. Push that boolean into the new list. Then check if the iterating variable is just a regular string, then just push that boolean into the list.

def flick_switch(lst):
	new_lst = []
	toggle = True
	for word in lst:
		if word == 'flick':
			toggle = not(toggle)
		new_lst.append(toggle)

	return new_lst

print(flick_switch(["flick", "flick", "flick", "flick", "flick"]))
print(flick_switch(["bird"]))
print(flick_switch(["flick", "mountain", "tiger", "flick", "dog"]))
print(flick_switch(["plane", "jar", "bed", "flick", "boat"]))
print(flick_switch(["flick"]))